app/bars/model/ValidateBankDetailsResponse.scala

```diff
@@ -36,7 +36,11 @@ final case class ValidateBankDetailsResponse(
   def isValid: Boolean =
     (accountNumberIsWellFormatted == Yes
       || accountNumberIsWellFormatted == Indeterminate) &&
-      sortCodeIsPresentOnEISCD == Yes
+      sortCodeIsPresentOnEISCD == Yes &&
+      !sortCodeSupportsDirectDebit.contains(No)
```

Comment:
Maybe call the supportsDirectDebit method here instead of repeating the same line of code? Easier if you change the method in the future, too.